api/v1alpha1/virtualmachine_types.go

```diff
@@ -185,6 +189,26 @@ type TCPSocketAction struct {
 	Host string `json:"host,omitempty"`
 }
 
+// The guest heartbeat status.
+type GuestHeartbeatStatus string
+
+// See govmomi.vim25.types.ManagedEntityStatus
+const (
+	// Intermittent heartbeat. May be due to guest load.
+	YellowStatus GuestHeartbeatStatus = "yellow"
+	// Guest operating system is responding normally.
+	GreenStatus  GuestHeartbeatStatus = "green"
+)
+
+// GuestHeartbeatAction describes an action based on the guest heartbeat.
+type GuestHeartbeatAction struct {
+	// MinimumStatus is the minimum heartbeat status that is considered successful.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=green
+	// +kubebuilder:validation:Enum=yellow;green
+	MinimumStatus GuestHeartbeatStatus `json:"minimumStatus,omitempty"`
```

Comment:
What do you think about calling it `ThresholdStatus` and adding a comment that says whenever the heartbeat status doesn't pass the threshold, the probe is considered to fail?

Reply:
That's fine